lib/libzpool/uzfs_mgmt.c

```diff
@@ -305,6 +325,7 @@ uzfs_open_dataset_init(const char *ds_name, zvol_state_t **z)
 	if (error)
 		goto disown_free;
 
+	/* Meta-object may not exist if dataset wasn't created by uzfs */
```

Comment:
If I understood, this PR is creating meta-object even if dataset wasn't created by uzfs.

Reply:
I think that no, although the function names in this area are rather confusing as Jeffry pointed out during our morning session so I might be wrong :)  meta object is created only in two cases:

 * uzfs_create_dataset - used by our tests
 * ioctl which calls zvol_create_cb and this calls create_meta - ioctl path

Reply:
If I'm not wrong, these are the only ways uzfs should be creating dataset.

Reply:
yes, I think so too.

Reply:
can we remove this comment, as in both the cases, we are creating metaobject.